runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/WindowIntoTranslation.java

```diff
@@ -64,10 +66,41 @@ public static WindowIntoPayload toProto(Window.Assign<?> transform, SdkComponent
         .build();
   }
 
-  public static WindowFn<?, ?> getWindowFn(WindowIntoPayload payload)
-      throws InvalidProtocolBufferException {
-    SdkFunctionSpec spec = payload.getWindowFn();
-    return WindowingStrategyTranslation.windowFnFromProto(spec);
+  public static WindowFn<?, ?> getWindowFn(AppliedPTransform<?, ?, ?> application) {
+    RunnerApi.PTransform transformProto;
+    try {
+      transformProto =
+          PTransformTranslation.toProto(
+              application,
+              Collections.<AppliedPTransform<?, ?, ?>>emptyList(),
+              SdkComponents.create());
+    } catch (IOException exc) {
+      throw new RuntimeException(exc);
+    }
+
+    checkArgument(
+        PTransformTranslation.WINDOW_TRANSFORM_URN.equals(transformProto.getSpec().getUrn()),
+        "Illegal attempt to extract %s from transform %s with name \"%s\" and URN \"%s\"",
+        Window.Assign.class.getSimpleName(),
+        application.getTransform(),
+        application.getFullName(),
+        transformProto.getSpec().getUrn());
+
+    WindowIntoPayload windowIntoPayload;
+    try {
+      windowIntoPayload = transformProto.getSpec().getParameter().unpack(WindowIntoPayload.class);
```

Comment:
Is this worth factoring into two methods, `getWindowIntoPayload(AppliedPTransform)` and then the `windowFnFromProto`? That seems to be the style we use elsewhere, and I think I generally like it.